Add Statsig analytics and log errors via Metrics Reporter for accept-reject flow

fisher-alice opened the pull request:
This PR adds Statsig event logging for when a user accepts or rejects AI tutor version file updates in Web Lab 2 accept-reject flow. It also logs errors via Metrics Reporter during the flow.

Note that in the JSON schema description included in the request, the model is asked to limit the code files to one language (html, css or js) across the entire files list (`html`, `css`, or `js` fences). The Statsig event's payload includes `numFiles`, `filesType` (limited to one), `levelPath`, and `appName`. Locally testing, the model has consistently followed this rule in contrast to when we previously asked the model to returning the file `id` and `folderId` of updated files.

## After update

User accepts AI Tutor version files:
<img width="1366" height="202" alt="Screenshot 2025-12-10 at 9 29 17 AM" src="https://github.com/user-attachments/assets/212bedcc-0dfc-4c91-a19f-8f4cd5a4273e" />

User rejects AI Tutor version files
<img width="1356" height="262" alt="Screenshot 2025-12-10 at 9 27 15 AM" src="https://github.com/user-attachments/assets/a1a19ea9-bcb1-47a6-827c-c4d69ed5a9de" />

Statsig events sent for:
- when code is generated (html, css, or js),
- user clicks on preview button,
- user enters file path in url bar in AI Tutor version view,
- user clicks on file tab in file browser or tabs bar (note that the event name has since been updated so these two event are now the same)

https://github.com/user-attachments/assets/131cad61-5431-429f-bcf9-d8941fbddb61


## Warning!!

We have entered Pixel Lock for Hour of AI! All merges to the `staging` branch from Dec 2 through Dec 12 must go through live change review and be deemed critical for supporting the Hour of AI. External contributions will not be accepted at this time.

For non-critical changes, please change your base to `staging-next` and delete this warning. We will merge `staging-next` into `staging` on Dec 15, 2025.

<!-- end warning -->


<!--
  A summary of the change, including any relevant background, motivation, and context.
  If relevant, include a description, screenshots, and/or video of the existing and new behavior.
-->


## Links
<!--  Links to relevant external resources.
      - spec docs, Jira tickets, related PRs, Honeybadger errors, etc.
-->

- Jira: https://codedotorg.atlassian.net/browse/AFL-290

## Testing story
<!--  Does your change include appropriate tests?
      - If so, please describe how the tests included in this PR are sufficient.
      - If not, please explain why this change does not need to be tested. -->

Tested locally and confirmed the statsig event logs in dev console.

## Deployment strategy
<!--  Any special steps required to deploy this, besides merging?
      - Are there database migrations or manual steps?
      - Does this require coordination with other teams or systems? -->



## Follow-up work
<!--  List any clean-up or technical debt that will be addressed in future work.
      - Include Jira links where applicable.
      - Are there any known limitations or improvements planned? -->



## Privacy
<!--  How does this change impact Student & Teacher privacy?
      - Does this collect, use, share, or modify PII, either new or existing? -->



## Security
<!--  How does this change impact our security surface-area?
      - Do API's touched have the right auth?
      - Do infra changes impact our attack surface or encryption?
-->



## Caching
<!--  How does this change impact caching behavior?
      - Are cache keys or invalidation strategies affected?
      - Will this require cache warming or invalidation after deployment? -->



## PR Creation Checklist:
<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy impacts have been documented
- [ ] Security impacts have been documented
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

fisher-alice committed:
add statsig events for accept-reject and log errors via metrics reporter

fisher-alice commented:
> Hey @fisher-alice - Thinking about this now. Wondering if we need an "AI Tutor Generated Code" event so we can look at what students do between the code being generated and accept/reject. Which then left we wondering if we should have events to know that a student:
> 
> * changed files in the code area
> * interacted with the preview area
>   I think that would give the more of the holistic picture of what are students doing. What do you think?

Thanks Dani! When the workspace is in AI Tutor version view, it is read-only so student cannot update project files during this state. However, we could send an event when students click on the preview eye icon button which is only rendered during AI Tutor version view. We could also send an event if students enter a URL in the preview while in AI Tutor version view. Lmk what you think.

fisher-alice commented:
> Hey @fisher-alice - Thinking about this now. Wondering if we need an "AI Tutor Generated Code" event so we can look at what students do between the code being generated and accept/reject. 

Wrt to an 'AI Tutor Generated Code' event, we can get this from the DB once we remove the accept-reject experiment flag and the `tutorMode` is either 'Build HTML', 'Build CSS', or 'Build JavaScript'. But lmk if you'd still like to add a statsig event as well. Thanks!

dmcavoy commented:
> > Hey @fisher-alice - Thinking about this now. Wondering if we need an "AI Tutor Generated Code" event so we can look at what students do between the code being generated and accept/reject.
> 
> Wrt to an 'AI Tutor Generated Code' event, we can get this from the DB once we remove the accept-reject experiment flag and the `tutorMode` is either 'Build HTML', 'Build CSS', or 'Build JavaScript'. But lmk if you'd still like to add a statsig event as well. Thanks!

I would love to have an event for this in statsig.

fisher-alice committed:
add more statsig events requested by product

fisher-alice committed:
Merge branch 'staging-next' into alice/accept-reject-logs

fisher-alice committed:
distinguish between isAiTutorVersionFile and isAiTutorVersion

fisher-alice commented:
> I would love to have an event for this in statsig.

Added 'AI Tutor Generated Code' event!

> Oh sorry by changed files I meant just looked at different files. Really want I was trying to get at here is the question of "What are students doing between code generated and code accepted/rejected? Do they actually view the code? Do they test the preview? Or do they just accept?"

Gotcha! I added the following events:
```
 AI_TUTOR_VERSION_FILE_TAB_CLICKED_IN_TABS_BAR:  'File Tab Clicked in File Tabs Bar in AI Tutor Version View',
  AI_TUTOR_VERSION_FILE_BROWSER_TAB_CLICKED_IN_FILE_BROWSER: 'File Browser Tab Clicked in File Browser in AI Tutor Version View',
  AI_TUTOR_VERSION_FILE_PREVIEW_BUTTON_CLICKED: 'AI Tutor Version File Preview Button Clicked',
  AI_TUTOR_VERSION_FILE_PREVIEWED_IN_URL_BAR:  'File Previewed in AI Tutor Version View via URL bar',
```

I also updated the PR description to include screencast videos of these added events. Lmk what you think.

fisher-alice commented:
@molly-moen - just re-requested your review since I made some non-trivial updates since you approved. Thanks!

dmcavoy reviewed (approved):
Thanks for the updates! With the combining of the file viewing events this meets logging needs for the V1.

fisher-alice committed:
combine events for file click and remove folder click event

fisher-alice committed:
Merge branch 'staging-next' into alice/accept-reject-logs
